server.c

```diff
@@ -0,0 +1,233 @@
+
+/* We want POSIX.1-2008 + XSI, i.e. SuSv4, features */
+#define _XOPEN_SOURCE 700
+/* Added on 2017-06-25:
+   If the C library can support 64-bit file sizes
+   and offsets, using the standard names,
+   these defines tell the C library to do so. */
+#define _LARGEFILE64_SOURCE
+#define _FILE_OFFSET_BITS 64
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <netdb.h>
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <ftw.h>
+
+#include <errno.h>
+
+#include <sys/stat.h>
+#include <sys/socket.h>
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#define MAX 80
+#define PORT 8080
```

Comment:
Create a common library for the server and client to use the same buffer size and port, instead of having to modify them in both files